packages/shared/ReactComponentStackFrame.js

```diff
@@ -131,6 +131,9 @@ export function describeNativeComponentFrame(
       } catch (x) {
         control = x;
       }
+      // TODO(luna): This will currently only throw if the function component
+      // tries to access React/ReactDOM/props. We should probably make this throw
+      // in simple components too
```

Comment:
I don't think I understand this comment. What do you mean? A component that doesn't use props or state or hooks? (How would we make such a function throw?)

Reply:
This code works by relying on the render function throwing (`fn()`) in this case, and then getting the error stack trace from that in production and places where the source isn't defined. However, sometimes the function won't actually throw, which means this code won't actually get the source location. Need to think about this more though.

Reply:
Agreed. I'm just not sure how we _could_ make a function that doesn't use props/state/hook throw. Maybe if we had a way to override `createElement` to make it throw? heh

Reply:
Yeah I'm not sure either. Putting a TODO to think about it at some point.